src/main/java/blusunrize/immersiveengineering/common/util/compat/OneProbeCompatModule.java

```diff
@@ -0,0 +1,356 @@
+/*
+ * BluSunrize, RobustProgram
+ * Copyright (c) 2020
+ *
+ * This code is licensed under "Blu's License of Common Sense"
+ * Details can be found in the license file in the root folder of this project
+ */
+
+package blusunrize.immersiveengineering.common.util.compat;
+
+import blusunrize.immersiveengineering.ImmersiveEngineering;
+import blusunrize.immersiveengineering.api.IEEnums.IOSideConfig;
+import blusunrize.immersiveengineering.api.Lib;
+import blusunrize.immersiveengineering.api.energy.immersiveflux.IFluxProvider;
+import blusunrize.immersiveengineering.api.energy.immersiveflux.IFluxReceiver;
+import blusunrize.immersiveengineering.common.blocks.IEBlockInterfaces;
+import blusunrize.immersiveengineering.common.blocks.generic.MultiblockPartTileEntity;
+import blusunrize.immersiveengineering.common.blocks.metal.SheetmetalTankTileEntity;
+import blusunrize.immersiveengineering.common.blocks.metal.TeslaCoilTileEntity;
+import com.google.common.base.Function;
+import mcjty.theoneprobe.Tools;
+import mcjty.theoneprobe.api.*;
+import mcjty.theoneprobe.config.Config;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.world.World;
+import net.minecraft.util.Direction;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.fml.InterModComms;
+
+import javax.annotation.Nullable;
+import java.util.function.Supplier;
+
+/**
+ * @author BluSunrize, Robustprogram - 3.1.2020
+ */
+public class OneProbeCompatModule extends IECompatModule implements Function<ITheOneProbe, Void>
+{
+	@Override
+	public void preInit()
+	{
+		Supplier<Function<ITheOneProbe, Void>> supplier = () -> this;
+		InterModComms.sendTo("theoneprobe", "getTheOneProbe", supplier);
+	}
+
+	@Override
+	public void registerRecipes() { }
+
+	@Override
+	public void init() { }
+
+	@Override
+	public void postInit() { }
+
+	@Nullable
+	@Override
+	public Void apply(@Nullable ITheOneProbe input)
+	{
+		EnergyInfoProvider energyInfo = new EnergyInfoProvider();
+		input.registerProvider(energyInfo);
+		input.registerProbeConfigProvider(energyInfo);
+		input.registerProvider(new ProcessProvider());
+		input.registerProvider(new TeslaCoilProvider());
+		input.registerProvider(new SideConfigProvider());
+		input.registerProvider(new FluidInfoProvider());
+		input.registerBlockDisplayOverride(new MultiblockDisplayOverride());
+		return null;
+	}
+
+	public static class FluidInfoProvider implements IProbeInfoProvider
+	{
+
+		@Override
+		public String getID()
+		{
+			return ImmersiveEngineering.MODID+":"+"FluidInfo";
+		}
+
+		@Override
+		public void addProbeInfo(
+			ProbeMode mode,
+			IProbeInfo probeInfo,
+			PlayerEntity player,
+			World world,
+			BlockState blockState,
+			IProbeHitData data
```

Comment:
Please condense these into fewer lines (two or three), this does not match out usual codestyle